packages/server/utils/analytics/analytics.ts

```diff
@@ -0,0 +1,101 @@
+import MeetingTemplate from '../../database/types/MeetingTemplate'
+import {TeamPromptResponse} from '../../postgres/queries/getTeamPromptResponsesByIds'
+import segment from '../segmentIo'
+import {createMeetingTemplateAnalyticsParams} from './helpers'
+import {SegmentAnalytics} from './segment/SegmentAnalytics'
+
+export type AnalyticsEvent = 'Meeting Started' | 'Meeting Joined' | 'Meeting Completed'
+
+/**
+ * Provides a unified inteface for sending all the analytics events
+ */
+class Analytics {
+  private segmentAnalytics: SegmentAnalytics
+
+  constructor() {
+    this.segmentAnalytics = new SegmentAnalytics(segment)
+  }
+
+  teamPromptEnd = (
+    completedMeeting: Meeting,
+    meetingMembers: MeetingMember[],
+    responses: TeamPromptResponse[]
+  ) => {
+    const userIdsResponses: Record<string, string> = responses.reduce(
+      (previous, response) => ({...previous, [response.userId]: response.plaintextContent}),
+      {}
+    )
+    meetingMembers.forEach((meetingMember) => {
+      const plaintextResponseContent = userIdsResponses[meetingMember.userId]
+      this.meetingEnd(meetingMember.userId, completedMeeting, meetingMembers, {
+        responseAdded: plaintextResponseContent && plaintextResponseContent.length > 0
+      })
+    })
+  }
+
+  checkInEnd = (completedMeeting: Meeting, meetingMembers: MeetingMember[]) => {
+    meetingMembers.forEach((meetingMember) =>
+      this.meetingEnd(meetingMember.userId, completedMeeting, meetingMembers)
+    )
+  }
+
+  retrospectiveEnd = (
+    completedMeeting: Meeting,
+    meetingMembers: MeetingMember[],
+    template: MeetingTemplate
+  ) => {
+    meetingMembers.forEach((meetingMember) =>
+      this.meetingEnd(
+        meetingMember.userId,
+        completedMeeting,
+        meetingMembers,
+        createMeetingTemplateAnalyticsParams(template)
+      )
+    )
+  }
+
+  sprintPokerEnd = (
+    completedMeeting: Meeting,
+    meetingMembers: MeetingMember[],
+    template: MeetingTemplate
+  ) => {
+    meetingMembers.forEach((meetingMember) =>
+      this.meetingEnd(
+        meetingMember.userId,
+        completedMeeting,
+        meetingMembers,
+        createMeetingTemplateAnalyticsParams(template)
+      )
+    )
+  }
+
+  private meetingEnd = (
+    userId: string,
+    completedMeeting: Meeting,
+    meetingMembers: MeetingMember[],
+    meetingSpecificProperties?: any
+  ) => {
+    const {facilitatorUserId, meetingNumber, meetingType, phases, teamId} = completedMeeting
+    const presentMemberUserIds = meetingMembers.map(({userId}) => userId)
+    const wasFacilitator = userId === facilitatorUserId
+    this.track(userId, 'Meeting Completed', {
+      hasIcebreaker: phases[0]?.phaseType === 'checkin',
+      // include wasFacilitator as a flag to handle 1 per meeting
+      wasFacilitator,
+      userIds: wasFacilitator ? presentMemberUserIds : undefined,
+      meetingType,
+      meetingNumber,
+      teamMembersCount: meetingMembers.length,
```

Comment:
+1 Can you just add a `//TODO: fix me` so we don't forget this

Reply:
done